tests/ha/single_pod_restart/test_data_pod_restart_api.py

```diff
@@ -1221,3 +1222,229 @@ def test_obj_ver_suspension_data_pod_restart(self):
         LOGGER.info("COMPLETED: Test to verify bucket versioning suspension before & after data "
                     "pod restart.")
+
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-45637")
+    def test_obj_overwrite_data_pod_restart(self):
+        """
+        Verify object overwrite before and after data pod restart
+        """
+        LOGGER.info("STARTED: Test to verify object overwrite before and after data pod restart")
+        file_size = HA_CFG["5gb_mpu_data"]["file_size"]
+        LOGGER.info("Step 1: Create bucket %s and perform upload of object size %s MB and "
+                    "Overwrite the object", self.bucket_name, file_size)
+        s3_data = {self.bucket_name: [self.object_name, file_size]}
+        resp = self.ha_api.object_overwrite_dnld(self.s3_test_obj, s3_data, iteration=1,
+                                                 random_size=False)
+        assert_utils.assert_true(resp[0], "Failure observed in overwrite method.")
+        for _, checksum in resp[1].items():
+            assert_utils.assert_equal(checksum[0], checksum[1],
+                                      f"Checksum doesn't match, Expected: {checksum[0]} "
+                                      f"Received: {checksum[1]}")
+        LOGGER.info("Step 1: Create bucket %s and perform upload of object size %s MB and "
+                    "Overwrite the object", self.bucket_name, file_size)
+
+        LOGGER.info("Step 2: Shutdown data pod with replica method and verify cluster & "
+                    "remaining pods status")
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+            master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+            delete_pod=[self.delete_pod], num_replica=self.num_replica - 1)
+        # Assert if empty dictionary
+        assert_utils.assert_true(resp[1], "Failed to shutdown/delete data pod")
+        pod_name = list(resp[1].keys())[0]
+        self.set_name = resp[1][pod_name]['deployment_name']
+        self.restore_pod = True
+        self.restore_method = resp[1][pod_name]['method']
+        assert_utils.assert_true(resp[0], "Cluster/Services status is not as expected")
+        LOGGER.info("Step 2: Successfully shutdown data pod %s. Verified cluster and "
+                    "services states are as expected & remaining pods status is online.", pod_name)
+
+        LOGGER.info("Step 3: Overwrite existing object %s of bucket %s.", self.object_name,
+                    self.bucket_name)
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("Step 3.1: Create new bucket and overwrite object in new bucket")
+            t_t = int(perf_counter_ns())
+            bucket_name = f"bucket-{t_t}"
+            object_name = f"object-{t_t}"
+            s3_data.update({bucket_name: [object_name, file_size]})
+        resp = self.ha_api.object_overwrite_dnld(self.s3_test_obj, s3_data, iteration=1,
```

Comment:
As you are passing new bucket to the overwrite method, it will first try to create it.
You will have to either create bucket inside dtm flag condition or you can call overwrite method inside dtm flag condition.

Reply:
As discussed, in case of dtm disable, we are adding new buckets to list,
if not, we are passing s3_data as it is, where we don't need to create new bucket and over-write will happen in already created buckets.